### PR TITLE
The christmas gift of... Invincibility?

### PR DESCRIPTION
**Pull request by X0-11**
:cl: XO-11
rscadd: A bunch of Christmas goodies
/:cl:

### CODE REVIEW

File: code/modules/halo/misc/admin_truce.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+/datum/admin_secret_item/fun_secret/enact_truce/can_execute(var/mob/user)
+	if(!ticker) return 0
+	return ..()
+
+/datum/admin_secret_item/fun_secret/enact_truce/proc/toggle_truce(var/on = 0)
+	for(var/mob/living/m in world)
+		if(on)
+			m.status_flags |= GODMODE
+		else
+			m.status_flags &= GODMODE
```
Comment (bloxgate) [inline code/modules/halo/misc/admin_truce.dm]:
```dm
m.status_flags &= ~GODMODE
```
We want to turn this off, so we need to zero the godmode bit

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 2]
```diff
@@ -43,14 +43,17 @@
 
 /mob/living/carbon/human/getHalLoss()
 	var/amount = 0
+	if(status_flags & GODMODE) return 0
 	for(var/obj/item/organ/external/E in organs)
 		amount += E.get_pain()
 	return amount * species.pain_mod
 
 /mob/living/carbon/human/setHalLoss(var/amount)
+	if(status_flags & GODMODE) return
```
Comment (bloxgate) [inline code/modules/mob/living/carbon/human/human_damage.dm]: `return 0`?

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 3]
```diff
@@ -372,7 +388,7 @@ This function restores all organs.
 				C.update_damage_description(damage_type)
 
 /mob/living/carbon/human/apply_damage(var/damage = 0, var/damagetype = BRUTE, var/def_zone = null, var/blocked = 0, var/damage_flags = 0, var/obj/used_weapon = null)
-
+	if(status_flags & GODMODE) return
```
Comment (bloxgate) [inline code/modules/mob/living/carbon/human/human_damage.dm:391]: Are you sure all of these are necessary? I was under the impression that enabling godmode pretty much already took care of this.

File: html/changelogs/.all_changelog.yml [hunk 4]
```diff
@@ -1750,3 +1750,9 @@ DO NOT EDIT THIS FILE BY HAND!  AUTOMATICALLY GENERATED BY ss13_genchangelog.py.
   - tweak: Attachment modifications
   - bugfix: Vehicles no longer suppress themselves
   - tweak: Lowered the sounds of overmap weapons
+2019-12-24:
```
Comment (bloxgate) [inline html/changelogs/.all_changelog.yml:1753]: Why is this in this PR?

**Review by bloxgate — CHANGES_REQUESTED** [review]

File: html/changelogs/.all_changelog.yml (hunk 4 above)
Comment (X0-11) [reply html/changelogs/.all_changelog.yml:1753]: very good question, I'm afraid I don't know. Perhaps an artifact from updating to main code at some point?

Comment (bloxgate) [reply html/changelogs/.all_changelog.yml:1753]: Perhaps. It's already present on the `alpha` branch, so it should be good to just replace your local copies of the CL files with those.

**Review by bloxgate — APPROVED** [review]

**Pull request merged**